nncf/data/dataset.py

```diff
@@ -81,6 +81,17 @@ def get_length(self) -> Optional[int]:
             return self._data_source.__len__()
         return None
 
+    def get_batch_size(self) -> Optional[int]:
+        """
+        Tries to fetch batch size of the underlying dataset.
+        :return: The value of batch_size or _batch_size attributes of the data_source if exist, and None otherwise.
+        """
+        if hasattr(self._data_source, "batch_size"):  # Torch dataloader
+            return self._data_source.batch_size
+        if hasattr(self._data_source, "_batch_size"):  # TF dataloader
+            return self._data_source._batch_size
+        return None
```

Comment:
This does not explain the behaviour in the case of the other data loaders (ONNX, OpenVINO) since this method was placed in a common part.

Reply:
I do not know dataloaders from ONNX or OpenVINO backends. AFAIK only Torch and TF are used or custom iterable objects, which is no way to generally support.
So here common representation for Torch and TF are being checked

Reply:
What about custom data loaders? For example, [from the accuracy checker](https://github.com/openvinotoolkit/open_model_zoo/blob/master/tools/accuracy_checker/openvino/tools/accuracy_checker/dataset.py#L77). 
I think that an attempt to somehow generalize such an entity as a `nncf.Dataset` is not applicable here since it works with any iterable `data_source`.

Reply:
For now NNCF assumes that `batch_size` is always 1 and does not check it. This could lead to inaccurate statistics collection.
The proposal supports the most common dataloaders (Torch and TF format). for batch size parameter extraction.
Do you have a better suggestion how NNCF can get batch size value?

Reply:
We may provide the new parameter for the `nncf.Dataset` class called `batch_size`, along with the `data_source`.
If we wanted to support only Torch and TF format, we would fix it as the `data_source` input, for example. Don't you agree?

Reply:
The most straightforward solution is to make this parameter optional and try to use it if provided. No guessing based on the `hasattr`. 
If there is a mismatch then dataloader should warn about this, not the NNCF.

Reply:
As it touches API I would ask @alexsu52 as well.

Reply:
And @AlexKoff88